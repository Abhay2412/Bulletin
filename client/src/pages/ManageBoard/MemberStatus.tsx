import React from "react";
import { Box } from "@mui/system";
import {
  Typography,
  Button,
  Select,
  MenuItem,
  SelectChangeEvent,
} from "@mui/material";

interface Props {
  name: string;
  email: string;
  role: string;
}

//Member status display on manage board - name, email, role and remove
export default function MemberStatus({ name, email, role }: Props) {
  const [newRole, setNewRole] = React.useState("");

  const handleChange = (event: SelectChangeEvent) => {
    setNewRole(event.target.value as string);
  };

  return (
    <Box display={"flex"} flexDirection={"row"} justifyContent="space-evenly">
      <Box
        width="30%"
        sx={{ border: "1px solid rgb(104, 57, 13)" }}
        alignItems="center"
        display={"flex"}
        justifyContent="center"
      >
        <Typography color={"primary"}>{name}</Typography>
      </Box>

      <Box
        width="40%"
        sx={{ border: "1px solid rgb(104, 57, 13)" }}
        alignItems="center"
        display={"flex"}
        justifyContent="center"
      >
        <Typography color={"primary"}>{email}</Typography>
      </Box>

      <Box
        width="25%"
        sx={{ border: "1px solid rgb(104, 57, 13)" }}
        alignItems="center"
        display={"flex"}
        justifyContent="center"
      >
        {/* <Typography color={"primary"}> */}
        <Select
          sx={{ width: "100%" }}
          // labelId="demo-simple-select-label"
          // id="demo-simple-select"
          variant="outlined"
          // value={role}
          color="primary"
          defaultValue={role}
          onChange={handleChange}
        >
          <MenuItem color="primary" value={"Admin"}>
            Admin
          </MenuItem>
          <MenuItem color='primary' value={"Member"}>Member</MenuItem>
        </Select>
        {/* {role} */}
        {/* </Typography> */}
      </Box>

      <Box
        width="5%"
        display={"flex"}
        flexDirection="row"
        alignItems={"center"}
      >
        <Button>
          <Typography fontWeight={"bold"} color="primary" variant="h6">
            X
          </Typography>
        </Button>
      </Box>
    </Box>
  );
}
