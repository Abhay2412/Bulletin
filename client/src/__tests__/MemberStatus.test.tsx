import React from "react";
import MemberStatus from "../pages/ManageBoard/MemberStatus";
import { configure, mount, ReactWrapper } from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

describe("Testing <MemberStatus /> Component", () => {
  const wrapper: ReactWrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <MemberStatus
          userBoardid=""
          name=""
          email=""
          role="Admin"
          id=""
          onDelete={() => {}}
        />
      </BrowserRouter>
    </Provider>
  );

  it("should render properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <MemberStatus
              userBoardid=""
              name=""
              email=""
              role="Admin"
              id=""
              onDelete={() => {}}
            />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should change member status dropdown", () => {
    wrapper
      .find("input")
      .find(".MuiSelect-nativeInput")
      .at(0)
      .simulate("change", {
        target: { value: "Member", name: "Member" },
      });

    expect(
      wrapper.find("input").find(".MuiSelect-nativeInput").get(0).props.value
    ).toEqual("Member");
  });
});
