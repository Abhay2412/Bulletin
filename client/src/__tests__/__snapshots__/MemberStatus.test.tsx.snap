// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <MemberStatus /> Component should render properly 1`] = `
<div>
  <div
    className="MuiBox-root css-ol10w6"
  >
    <div
      className="MuiBox-root css-9xqsl1"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
      >
        
      </p>
    </div>
    <div
      className="MuiBox-root css-ow002a"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
      >
        
      </p>
    </div>
    <div
      className="MuiBox-root css-1bvyfe8"
    >
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary  css-bx5laj-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        onClick={[Function]}
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="manage-board-member-status-select"
          className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="manage-board-member-status-select"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Admin
        </div>
        <input
          aria-hidden={true}
          className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          onAnimationStart={[Function]}
          onChange={[Function]}
          tabIndex={-1}
          value="Admin"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-hdw1oc"
          >
            <span
              className="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiBox-root css-19wx6p5"
    >
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-1kor5r7-MuiTypography-root"
        >
          X
        </h6>
      </button>
    </div>
  </div>
</div>
`;
